lib/spack/spack/cmd/create.py

```diff
@@ -120,39 +138,78 @@ def setup_parser(subparser):
-
-        config_lines = ((r'/configure$',      'autotools', autotools),
-                        (r'/CMakeLists.txt$', 'cmake',     cmake),
-                        (r'/setup.py$',       'python',    python),
-                        (r'/NAMESPACE$',      'r',         r))
-
-        # Peek inside the tarball.
-        tar = which('tar')
-        output = tar(
-            "--exclude=*/*/*", "-tf", stage.archive_file, output=str)
-        lines = output.split("\n")
-
-        # Set the configure line to the one that matched.
-        for pattern, bs, cl in config_lines:
+        """Try to guess the type of build system used by the project. Set the
+        appropriate default configure, build, and install instructions."""
+
+        # Default configure instructions
+        configureDict = {
+            'autotools': "configure('--prefix={0}'.format(prefix))",
+            'cmake':     "cmake('.', *std_cmake_args)",
+            'scons':     "",
+            'python':    "",
+            'r':         "",
+            'unknown':   "# FIXME: Unknown build system"
+        }
+
+        # Default build instructions
+        buildDict = {
+            'autotools': "make()",
+            'cmake':     "make()",
+            'scons':     "scons('prefix={0}'.format(prefix))",
+            'python':    "",
+            'r':         "",
+            'unknown':   "make()",
+        }
+
+        # Default install instructions
+        installDict = {
+            'autotools': "make('install')",
+            'cmake':     "make('install')",
+            'scons':     "scons('install')",
+            'python':    "python('setup.py', 'install', " +
+                         "'--prefix={0}'.format(prefix))",
+            'r':         "R('CMD', 'INSTALL', '--library={0}'.format(" +
+                         "self.module.r_lib_dir), self.stage.archive_file)",
+            'unknown':   "make('install')",
+        }
+
+        # A list of clues that give us an idea of the build system a package
+        # uses. If the regular expression matches a file contained in the
+        # archive, the corresponding build system is assumed.
+        clues = [
+            (r'/configure$',      'autotools'),
+            (r'/CMakeLists.txt$', 'cmake'),
+            (r'/SConstruct$',     'scons'),
+            (r'/setup.py$',       'python'),
+            (r'/NAMESPACE$',      'r')
+        ]
+
+        # Peek inside the compressed file.
+        output = ''
+        if stage.archive_file.endswith(('.tar', '.tar.gz', '.tar.bz2',
+                                        '.tgz', '.tbz2')):
+            tar    = which('tar')
+            output = tar('--exclude=*/*/*', '-tf',
+                         stage.archive_file, output=str)
+        elif stage.archive_file.endswith('.gz'):
+            gunzip = which('gunzip')
+            output = gunzip('-l', stage.archive_file, output=str)
```

Comment:
Can .gz files contain directories? Gunzip does have the option to list files, but that might just be for .tar.gz files...